File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.py
from collections import Counter

def solution(k, tangerine):
    count = 0
    counter = Counter(tangerine)
    
    for i, (_, n) in enumerate(counter.most_common()):
        count += n
        if count >= k:
            return i + 1
    
    return len(counter)

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.js
function solution(k, tangerine) {
    let count = 0;
    const counter = {};
    
    tangerine.forEach(item => {
        counter[item] = (counter[item] || 0) + 1;
    })
    
    const sorted_counter = Object.entries(counter).sort(([, a], [, b]) => b - a);
    
    for (let i = 0; i < sorted_counter.length; i++) {
        count += sorted_counter[i][1];
        if (count >= k) return i + 1;
    }
    
    return sorted_counter.length;
}